server/src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java

```diff
@@ -79,7 +76,7 @@ public MappingLookup mappers() {
     }
 
     public ParsedDocument parse(SourceToParse source) throws MapperParsingException {
```

Comment:
This method is used in a ton of tests, and one single place where it makes sense, where we create an empty document mapper once an empty doc gets indexed in an index that has no mappings yet. Maybe we should deprecate it, yet the fact that document mapper includes a document parser makes this very convenient to use, which should not be said about something we want to remove.

Reply:
IN general, there are a ton of usages of DocumentMapper#parse in tests that we should probably look at replacing.